backend/layers/business/business.py

```diff
@@ -143,33 +143,10 @@ def get_published_collection_version(self, collection_id: CollectionId) -> Optio
         """
         return self.database_provider.get_collection_mapped_version(collection_id)
 
-    def get_published_collection_version__discover_api(
+    def get_published_collection_version_with_datasets(
```

Comment:
I don't see enough of a difference between this and `get_collection_version` on line:165 to warrant a new function.